tests/components/hyperion/test_light.py

```diff
@@ -0,0 +1,430 @@
+        SERVICE_TURN_ON,
+        {ATTR_ENTITY_ID: TEST_ENTITY_ID, ATTR_BRIGHTNESS: brightness},
+        blocking=True,
+    )
+
+    assert client.async_send_set_adjustment.call_args == call(
+        **{const.KEY_ADJUSTMENT: {const.KEY_BRIGHTNESS: 100}}
+    )
+    assert client.async_send_set_color.call_args == call(
+        **{
+            const.KEY_PRIORITY: TEST_PRIORITY,
+            const.KEY_COLOR: (0, 255, 255),
+            const.KEY_ORIGIN: hyperion_light.DEFAULT_ORIGIN,
+        }
+    )
+    client.adjustment = [{const.KEY_BRIGHTNESS: 100}]
+    call_registered_callback(client, "adjustment-update")
+    entity_state = hass.states.get(TEST_ENTITY_ID)
+    assert entity_state.attributes["brightness"] == brightness
+
+    # On (=), 100% (=), V4L (!), [0,255,255] (=)
+    effect = const.KEY_COMPONENTID_EXTERNAL_SOURCES[2]  # V4L
+    client.async_send_clear = CoroutineMock(return_value=True)
+    client.async_send_set_component = CoroutineMock(return_value=True)
+    await hass.services.async_call(
+        DOMAIN,
+        SERVICE_TURN_ON,
+        {ATTR_ENTITY_ID: TEST_ENTITY_ID, ATTR_EFFECT: effect},
+        blocking=True,
+    )
+
+    assert client.async_send_clear.call_args == call(
+        **{const.KEY_PRIORITY: TEST_PRIORITY}
+    )
+    assert client.async_send_set_component.call_args_list == [
+        call(
+            **{
+                const.KEY_COMPONENTSTATE: {
+                    const.KEY_COMPONENT: const.KEY_COMPONENTID_EXTERNAL_SOURCES[0],
+                    const.KEY_STATE: False,
+                }
+            }
+        ),
+        call(
+            **{
+                const.KEY_COMPONENTSTATE: {
+                    const.KEY_COMPONENT: const.KEY_COMPONENTID_EXTERNAL_SOURCES[1],
+                    const.KEY_STATE: False,
+                }
+            }
+        ),
+        call(
+            **{
+                const.KEY_COMPONENTSTATE: {
+                    const.KEY_COMPONENT: const.KEY_COMPONENTID_EXTERNAL_SOURCES[2],
+                    const.KEY_STATE: True,
+                }
+            }
+        ),
+    ]
+    client.visible_priority = {const.KEY_COMPONENTID: effect}
+    call_registered_callback(client, "priorities-update")
+    entity_state = hass.states.get(TEST_ENTITY_ID)
+    assert entity_state.attributes["icon"] == hyperion_light.ICON_EXTERNAL_SOURCE
+    assert entity_state.attributes["effect"] == effect
+
+    # On (=), 100% (=), "Warm Blobs" (!), [0,255,255] (=)
+    effect = "Warm Blobs"
+    client.async_send_clear = CoroutineMock(return_value=True)
+    client.async_send_set_effect = CoroutineMock(return_value=True)
+
+    await hass.services.async_call(
+        DOMAIN,
+        SERVICE_TURN_ON,
+        {ATTR_ENTITY_ID: TEST_ENTITY_ID, ATTR_EFFECT: effect},
+        blocking=True,
+    )
+
+    assert client.async_send_clear.call_args == call(
+        **{const.KEY_PRIORITY: TEST_PRIORITY}
+    )
+    assert client.async_send_set_effect.call_args == call(
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6233107)